scripts/bosh-cli.sh

```diff
@@ -1,53 +1,114 @@
 #!/bin/bash
+set -euo pipefail
 
-set -eu
+dir="$(cd "$(dirname "$0")" && pwd)"
+pac_url='http://localhost:9191/proxy.pac'
+tunnel_mux='/tmp/bosh-ssh-tunnel.mux'
+pacserverpid=''
 
-SSH_PATH=${SSH_PATH:-"/Users/${USER}/.ssh/id_rsa"}
+if hash networksetup &> /dev/null; then
+  old_auto_proxy_settings=$(networksetup -getautoproxyurl Wi-Fi)
+  old_auto_proxy_url=$(awk '/^URL:/ { print $2 }' <<< "$old_auto_proxy_settings")
+  old_auto_proxy_enabled=$(awk '/^Enabled:/ { print $2 }' <<< "$old_auto_proxy_settings")
+fi
+function cleanup () {
+  if [ -n "$pacserverpid" ]; then
+    echo 'Stopping PAC server'
+    kill "$pacserverpid" &>/dev/null || true
+  fi
 
-USER_ID_RSA="$(base64 "${SSH_PATH}")"
-export USER_ID_RSA
+  echo 'Closing SSH tunnel'
+  ssh -S "$tunnel_mux" -O exit a-destination &>/dev/null || true
 
-BOSH_CA_CERT="$(aws s3 cp "s3://gds-paas-${DEPLOY_ENV}-state/bosh-CA.crt" -)"
-export BOSH_CA_CERT
+  # Avoid keeping sensitive tokens in bosh config when we don't need them.
+  # This will mean we have to sign in to bosh every time we run this script.
+  rm -f ~/.bosh/config
+
+  if hash networksetup &> /dev/null; then
+    echo 'Restoring network settings'
+    networksetup -setautoproxyurl Wi-Fi "$old_auto_proxy_url" &>/dev/null || true
+    if [ "$old_auto_proxy_enabled" == 'No' ]; then
+      networksetup -setautoproxystate Wi-Fi off
+    fi
+  fi
+}
+
+trap cleanup EXIT
+
+echo 'Starting PAC server'
+ruby "$dir/../proxy/pacserver.rb" &>/dev/null &
+pacserverpid="$!"
 
+if hash networksetup &> /dev/null && [[ "$pac_url" != "$old_auto_proxy_url" || "$old_auto_proxy_enabled" == "No" ]]; then
+  echo
+  echo 'Your system network settings need to be changed to allow your browser to access UAA over our SSH tunnel.'
+  echo 'Your current settings are: '
+  echo "$old_auto_proxy_settings"
+  echo
+  while true; do
+    read -r -p 'Would you like these to be changed? ' yn
```

Comment:
Should include the possible responses in the prompt `Would you like these to be changed? (y/n)`

Reply:
Done